tests/test_platform_utils.py

```diff
@@ -0,0 +1,200 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Rose. If not, see <http://www.gnu.org/licenses/>.
+# -----------------------------------------------------------------------------
+"""Tests for platform utils module:
+"""
+
+import os
+import pytest
+
+from pathlib import Path
+from shutil import rmtree
+from subprocess import run
+from uuid import uuid4
+
+from cylc.rose.platform_utils import (
+    get_platform_from_task_def,
+    get_platforms_from_task_jobs
+)
+
+from cylc.flow.cfgspec.globalcfg import SPEC
+from cylc.flow.parsec.config import ParsecConfig
+
+MOCK_GLBL_CFG = (
+    'cylc.flow.platforms.glbl_cfg',
+    '''
+    [platforms]
+        [[localhost]]
+            hosts = localhost
+        [[milk]]
+            hosts = milk
+        [[dairy]]
+            hosts = cheese, eggs
+    ''')
+
+
+@pytest.fixture
+def mock_glbl_cfg(tmp_path, monkeypatch):
+    """A Pytest fixture for fiddling global config values.
+
+    * Hacks the specified `glbl_cfg` object.
+    * Can be called multiple times within a test function.
+
+    Args:
+        pypath (str):
+            The python-like path to the global configuation object you want
+            to fiddle.
+            E.G. if you want to hack the `glbl_cfg` in
+            `cylc.flow.scheduler` you would provide
+            `cylc.flow.scheduler.glbl_cfg`
+        global_config (str):
+            The globlal configuration as a multi-line string.
+
+    Example:
+        Change the value of `UTC mode` in the global config as seen from
+        `the scheduler` module.
+
+        def test_something(mock_glbl_cfg):
+            mock_glbl_cfg(
+                'cylc.flow.scheduler.glbl_cfg',
+                '''
+                    [scheduler]
+                        UTC mode = True
+                '''
+            )
+
+    """
+    # TODO: modify Parsec so we can use StringIO rather than a temp file.
+    def _mock(pypath, global_config):
+        nonlocal tmp_path, monkeypatch
+        global_config_path = tmp_path / 'global.cylc'
+        global_config_path.write_text(global_config)
+        glbl_cfg = ParsecConfig(SPEC)
+        glbl_cfg.loadcfg(global_config_path)
+
+        def _inner(cached=False):
+            nonlocal glbl_cfg
+            return glbl_cfg
+
+        monkeypatch.setattr(pypath, _inner)
+
+    yield _mock
+    rmtree(tmp_path)
+
+
+@pytest.fixture(scope='session')
+def fake_flow():
+    """Set up enough of an installed flow for tests in module.
+
+    1. Set up an installed ``flow.cylc`` config file.
+    2. Set up a fake flow database in ``.service/db``.
+
+    Returns:
+        flow_name: Name of fake workflow.
+        flow_path: Path at which fake workflow is installed.
+    """
+    # Set up a config
+    flow_name: str = f'cylc-rose-platform-utils-test-{str(uuid4())[:6]}'
+    flow_path: Path
+    flow_path = Path(os.path.expandvars('$HOME/cylc-run')) / flow_name
```

Comment:
(No need for a type annotation here, as mypy can infer the type for a simple case like this)